TESTS/mbedmicro-rtos-mbed/timer/main.cpp

```diff
@@ -0,0 +1,872 @@
+
+using namespace utest::v1;
+
+#define US_PER_SEC 1000000
+
+/* NOTE: NRF51_DK platform has 32kHz timer clock which
+ *       gives resolution of 31.25us/clock tick.
+ *       Therefore for this board we need a huge delta
+ *       for the timing tests. */
+
+/* Macro to define delta based on system clock frequency.
+ * Assume that min. delta value is equal to 5 us.
+ *
+ * Calculate number of us per system tick. Then multiple calculated value by
+ * 10 and add 5. Truncate calculated delta to decimal value by casting to int.
+ *
+ * For 1 MHz clock delta is equal to 15 us.
+ *
+ * EXAMPLE K64F
+ * system clock frequency = 120 000 000 Hz (120 MHz) => 1 tick = ~0.083 us.
+ * delta = truncate(0.083 * 10 + 5) = truncate(5.83) = 5 us.
+ *
+ * EXAMPLE NRF51_DK
+ * system clock frequency = 32 678 Hz (~32 kHz) => 1 tick = 31.25 us.
+ * delta = truncate(31.25 * 10 + 5) = truncate(317.5) = 317 us.
+ */
+#define DELTA_SYS_CLK_US ((int)((1.0f / ((float)osKernelGetSysTimerFreq() / US_PER_SEC) ) * 10 + 5))
+
+#define DELTA_US DELTA_SYS_CLK_US
+#define DELTA_S  ((float)DELTA_SYS_CLK_US/US_PER_SEC)
+
+static Timer *p_timer;
+
+/* Global variable used to simulate passage of time
+ * in case when timer which uses user ticker is tested.
+ */
+static uint32_t curr_ticker_us_val;
+
+/* User ticker interface function. */
+static void my_interface_init()
+{
+    /* do nothing. */
+}
+
+/* User ticker interface function - only this
+ * ticker interface function is used by Timer API. */
+static uint32_t my_ticker_read(void)
+{
+    /* Simulate elapsed time. */
+    return curr_ticker_us_val;
+}
+
+/* User ticker interface function. */
+static void my_disable_interrupt(void)
+{
+    /* do nothing. */
+}
+
+/* User ticker interface function. */
+static void my_clear_interrupt(void)
+{
+    /* do nothing. */
+}
+
+/* User ticker interface function. */
+static void my_set_interrupt(timestamp_t timestamp)
+{
+    /* do nothing. */
+}
+
+/* User ticker interface function. */
+static void my_fire_interrupt(void)
+{
+    /* do nothing. */
+}
+
+/* User ticker event queue. */
+static ticker_event_queue_t my_events = { 0 };
+
+/* User ticker interface data. */
+static const ticker_interface_t us_interface = {
+    .init = my_interface_init,
+    .read = my_ticker_read, /* Only this function is used by the Timer. */
+    .disable_interrupt = my_disable_interrupt,
+    .clear_interrupt = my_clear_interrupt,
+    .set_interrupt = my_set_interrupt,
+    .fire_interrupt = my_fire_interrupt,
+};
+
+/* User ticker data structure. */
+static const ticker_data_t us_data = {
+    .interface = &us_interface,
+    .queue = &my_events
+};
+
+/* Function which returns user ticker data. */
+const ticker_data_t* get_user_ticker_data(void)
+{
+    return &us_data;
+}
+
+/* Initialisation of the Timer object which uses
+ * ticker data provided by the user.
+ *
+ * */
+utest::v1::status_t timer_user_ticker_setup_h(const Case *const source, const size_t index_of_case)
```

Comment:
What the suffix `_h` stands for ? `_handler` ? If so then given there is no such suffix define by `greentea_*_handler` I suggest to use the `_handler` suffix instead of its shortened form.